packages/core/src/tracing/instrumentation/cloud/aws-sdk/v3/index.js

```diff
@@ -81,13 +82,21 @@ function shimSmithySend(originalSend) {
     const command = smithySendArgs[0];
 
     if (isActive) {
-      const awsProduct = operationMap[command.constructor.name];
+      const serviceId = self.config && self.config.serviceId;
+      let awsProduct = serviceId && awsProducts.find(aws => aws.spanName === serviceId.toLowerCase());
 
-      if (awsProduct) {
+      if (awsProduct && awsProduct.supportsOperation(command.constructor.name)) {
         return awsProduct.instrumentedSmithySend(self, originalSend, smithySendArgs);
+      } else {
+        // This code will be removed in version 3.x release.
+        // We are keeping it here for now to avoid any unknown problems.
```

Comment:
Following up from the earlier discussion on this comment.
> Absolutely! Our current plan is to remove the 'else' section within version @instana/core@3.x.
> We're optimistic that by that time, we'll have successfully transitioned all other AWS SDK v3
> instrumentations to this new approach. [ref](https://instana.kanbanize.com/ctrl_board/56/cards/96212/details/).

That makes sense. Great that you have noted this down as a major version refactoring in the Kanbanize card.

_nitpick:_ How about we make this a bit more explicit here in the comment?

```suggestion
        // This code can be removed once all AWS SDK v3 instrumentations have been refactored to use the new approach
        // introduced in https://github.com/instana/nodejs/pull/838 for kinesis. That is: Do not use an explicit
        // operationsInfo/operationsMap map that restricts the traced operations to a subset of possible operations, but
        // instead allow _all_ operations to be traced, using the operation name from `command.constructor.name` for
        // span.data.$spanName.op. We plan to finish this refactoring before or with the next major release (3.x) of the
        // @instana packages.
```

I know, this is much more verbose, but maybe our future selves will find it easier to understand.

Reply:
Incorporated the suggestion 💯